src/targets/Logary.Targets.Stackdriver/Target_Stackdriver.fs

```diff
@@ -0,0 +1,173 @@
+                                                                  "namespace_id", ns
+                                                                  "instance_id", instance
+                                                                  "pod_id", pod
+                                                                  "container_name", container
+                                                                  "zone", zone ] |> dict
+  let resourceName = function
+  | ComputeInstance _ -> "gce_instance"
+  | Container _ -> "container"
+  
+  let mkMonitoredResource project resourceType =
+    let r = MonitoredResource(Type = resourceName resourceType)
+    r.Labels.Add(mkLabels project resourceType)
+    r
+
+  let writeRequests (conf : StackdriverConf) (state : State) (reqs : TargetMessage[]) =
+      let messages = ResizeArray<_>() // Updated by the loop.
+      let acks = ResizeArray<_>() // updated by the loop
+      
+      for m in reqs do
+        match m with
+        | Log (message, ack) ->
+          // Write and save the ack for later.
+          acks.Add ack
+          messages.Add <| write message
+        | Flush (ackCh, nack) -> ()
+
+      let writer = Job.Scheduler.isolate <| fun _ -> state.logger.WriteLogEntries(state.logName, state.resource, conf.labels, messages)   
+      let handler = Job.tryIn writer Job.result (Job.raises)
+      handler >>= fun res -> Job.conIgnore acks 
+  
+  let createState (conf : StackdriverConf) = 
+    { logger = LoggingServiceV2Client.Create()
+      resource = mkMonitoredResource conf.projectId conf.resource
+      logName = sprintf "projects/%s/logs/%s" conf.projectId conf.logId }
+
+  let dispose runtime state ackVar = 
+    // do! Job.Scheduler.isolate (fun _ -> (state :> IDisposable).Dispose())
+    ackVar *<= () :> Job<_>
+
+  // This is the main entry point of the target. It returns a Job<unit>
+  // and as such doesn't have side effects until the Job is started.
+  let loop (conf : StackdriverConf) // the conf is specific to your target
+           (runtime : RuntimeInfo) // this one,
+           (requests : RingBuffer<_>) // this one, and,
+           (shutdown : Ch<_>) = // this one should always be taken in this order
+
+    let rec loop (state : State) : Job<unit> =
+      // Alt.choose will pick the channel/alternative that first gives a value
+      Alt.choose [
+        // When you get the shutdown value, you need to dispose of your resources
+        // off of Hopac's execution context (see Scheduler.isolate below) and
+        // then send a unit to the ack channel, to tell the requester that
+        // you're done disposing.
+        shutdown ^=> dispose runtime state
+
+        // The ring buffer will fill up with messages that you can then consume below.
+        // There's a specific ring buffer for each target.
+        RingBuffer.takeBatch conf.batchSize requests ^=> (fun requests -> writeRequests conf state requests >>= fun res -> loop state)
```

Comment:
Thanks! That was a hairy one.  I ~~stole~~ borrowed a bunch of code from the internal file appender target to see how to handle batches.